src/ui/components/AppModal.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { ReactNode, Suspense, useEffect } from "react";
+
+import { hideModal } from "ui/actions/app";
+import { getLoadingFinished, getModal } from "ui/reducers/app";
+import { useAppDispatch, useAppSelector } from "ui/setup/hooks";
+import { ModalType } from "ui/state/app";
+
+import LoginModal from "./shared/LoginModal";
+import LoomModal from "./shared/LoomModal";
+import PassportDismissModal from "./shared/Modals/PassportDismissModal";
+import RenameReplayModal from "./shared/Modals/RenameReplayModal";
+import NewAttachment from "./shared/NewAttachment";
+
+const LaunchBrowserModal = React.lazy(() => import("./shared/LaunchBrowserModal"));
+const UserSettingsModal = React.lazy(() => import("./shared/UserSettingsModal"));
+const SharingModal = React.lazy(() => import("./shared/SharingModal"));
+
+export function AppModal() {
```

Comment:
This just seemed easier to maintain as its own component rather than one nested within another component file.